All bump-version commands are failing

### Environment Details

Please indicate the following details about the environment in which you found the bug:

* DeepEcho version: 0.7.0

### Error Description

When running any make commands that need to run bump-my-version, we get the following error

```bash
bump-my-version bump candidate --no-tag
Traceback (most recent call last):
  File "/Users/~~~~~/anaconda3/envs/deepecho-311/bin/bump-my-version", line 5, in <module>
    from bumpversion.cli import cli
  File "/Users/~~~~~~/anaconda3/envs/deepecho-311/lib/python3.11/site-packages/bumpversion/cli.py", line 13, in <module>
    from bumpversion.click_config import config_option
  File "/Users/~~~~~~/anaconda3/envs/deepecho-311/lib/python3.11/site-packages/bumpversion/click_config.py", line 10, in <module>
    from click.decorators import FC, _param_memo  # noqa: PLC2701
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: cannot import name 'FC' from 'click.decorators' (/Users/~~~~~~~~/anaconda3/envs/deepecho-311/lib/python3.11/site-packages/click/decorators.py)
make: *** [bumpversion-candidate] Error 1
```

This is because the newer version of `bump-my-version` need a new version of click, but `rundoc` requires click < 8.0.

### Steps to reproduce

Run the following

```bash
make bumpversion-candidate
```

### Additional context
The `rundoc` library is only used in our [readme](https://github.com/sdv-dev/DeepEcho/blob/main/.github/workflows/readme.yml) workflow. We can just remove it from the pyproject.toml since it isn't used for anything else and install it directly in the workflow. Then we can install the latest version of click. 

We also need to uncap twine in order for our `make release` command to work. There is a bug in the older versions.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -50,7 +50,6 @@ test = [
     'pytest-cov>=2.6.0',
     'pytest-rerunfailures>=10.3,<15',
     'jupyter>=1.0.0,<2',
-    'rundoc>=0.4.3,<0.5',
     'pytest-runner >= 2.11.1',
     'tomli>=2.0.0,<3',
 ]
@@ -60,7 +59,7 @@ dev = [
     # general
     'setuptools<70',
     'build>=1.0.0,<2',
-    'bump-my-version>=0.18.3,<1',
+    'bump-my-version>=0.18.3',
     'pip>=9.0.1',
     'watchdog>=1.0.1,<5',
 
@@ -68,7 +67,7 @@ dev = [
     'ruff>=0.3.2,<1',
 
     # distribute on PyPI
-    'twine>=1.10.0,<4',
+    'twine>=1.10.0',
     'wheel>=0.30.0',
 
     # Advanced testing
@@ -78,6 +77,7 @@ dev = [
     # Invoking test commands
     'invoke'
 ]
+readme = ['rundoc>=0.4.3,<0.5',]
 
 [tool.setuptools]
 include-package-data = true

